magma/backend/coreir_.py

```diff
@@ -96,21 +96,22 @@ def wrap_if_named_type(self, port, definition):
             return True
         if not port.is_input():
             return False
+        value = port.value()
         if not (isinstance(port, (AsyncReset, AsyncResetN, Clock)) or
-                isinstance(port.trace(), (AsyncReset, AsyncResetN, Clock))):
+                isinstance(value, (AsyncReset, AsyncResetN, Clock))):
             return False
-        value = port.trace()
-        if value is None or issubclass(type(value), type(port).flip()):
+        undirected_t = type(port).qualify(Direction.Undirected)
+        if value is None or issubclass(type(value), undirected_t):
             return False
         if isinstance(port, (AsyncReset, AsyncResetN, Clock)):
-            T = type(port).flip()
+            T = Out(type(port))
         else:
-            T = type(value).flip()
+            T = In(type(value))
 
         with definition.open():
             port.unwire(value)
             inst = self.define_wrap(T, type(port), type(value))()
-            wire(getattr(inst, "in"), convertbit(value, type(port)))
+            wire(convertbit(value, type(port)), getattr(inst, "in"))
             wire(inst.out, convertbit(port, type(value)))
```

Comment:
Some bug fixes that came up when I started wiring clock signals to unused for the inline verilog logic